gm4_mountaineering/data/gm4_mountaineering/functions/main.mcfunction

```diff
@@ -0,0 +1,15 @@
+execute as @a[gamemode=!spectator] at @s run function gm4_mountaineering:player_main
+
+# Initiate crampons when they are first equipped with a tag
+tag @a[tag=!gm4_mountaineering_using_crampons,predicate=gm4_mountaineering:wearing_crampons] add gm4_mountaineering_using_crampons
+
+# Clear crampon tag when crampons are removed
+tag @a[predicate=!gm4_mountaineering:wearing_crampons,tag=gm4_mountaineering_using_crampons] remove gm4_mountaineering_using_crampons
+
+# Reset the jump counter for use in the poles jump boost cooldown
+scoreboard players reset @a gm4_mountaineering_jump_counter
+
+# Apply damage to any climbing shulkers
+execute as @e[type=minecraft:shulker,tag=gm4_mountaineering_climb_assist] at @s run function gm4_mountaineering:climbing_effects/damage_shulkers
```

Comment:
While this method works, it might be better to kill shulkers when there are no players with the `gm4_mountaineering_using_crampons` tag are standing on it. This is just a suggestion of how it might be improved though, you'll have to test some methods out yourself